common/rpc-models.go

```diff
@@ -37,6 +39,71 @@ func (c *RpcCmd) Parse(s string) error {
 	return err
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// ResourceString represents a source or dest string, that can have
+// three parts: the main part, a sas, and extra query parameters that are not part of the sas.
+type ResourceString struct {
+	Value      string
```

Comment:
Do we have tests for this?

Reply:
No (other than implicitly testing it in the existing enumeration tests - although some of those will be calling the enumerators directly and so missing this).  I don't have any clear ideas in my head on how to add extra tests.  Do you have any suggestions?

Reply:
I'll merge this now, but can add tests separately if we can come up with thoughts on how to do so.